### PR TITLE
Roblox Presence - Added some pages that were previously marked as "Unknown page", cleaned up code

### PR DESCRIPTION
**Pull request by YJJcoolcool**
Some pages, such as the Develop page were previously shown as "Unknown Page" on Discord.
Sometimes the URL also showed a "#" (like "/games#/") which caused the presence to stop working, so I also fixed that.
I also cleaned up some of the code and fixed the errors and warnings.

Proof of it working:
image
image

### CODE REVIEW

File: websites/R/Roblox/dist/metadata.json [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
-    "www.roblox.com",
-    "web.roblox.com"
-  ],
-  "version": "2.3.5",
-  "logo": "https://i.imgur.com/O8T2o2j.png",
-  "thumbnail": "https://i.imgur.com/pufi4q1.png",
+  "category": "games",
+  "logo": "https://proxy.duckduckgo.com/iu/?u=https://i.imgur.com/75dFVYB.png",
+  "thumbnail": "https://proxy.duckduckgo.com/iu/?u=https://i.imgur.com/pufi4q1.png",
+  "version": "2.4.0",
```
Comment (StrikerFRFX) [inline websites/R/Roblox/dist/metadata.json:11]: bump by +1 on the last number

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]

**Comment by YJJcoolcool** [comment]
@Bas950 I resolved the conflicts and removed the DuckDuckGo proxy for the image in metadata.json. I also updated the version number.

**Comment by StrikerFRFX** [comment]
> @Bas950 I resolved the conflicts and removed the DuckDuckGo proxy for the image in metadata.json. I also updated the version number.

no need to @ us, we check github.

File: websites/R/Roblox/presence.ts [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
+  messageTab,
+  friendsTab,
+  inventoryTab,
+  groupName,
+  groupTab,
+  gameTab;
+
+const gameName = <HTMLHeadingElement>document.querySelector(
   "div.game-calls-to-action > div.game-title-container > h2"
 );
```
Comment (Bas950) [inline websites/R/Roblox/presence.ts:16]: move this up and merge with other const, but I have a feeling this one should've been set in updatedata event?

**Review by Bas950 — CHANGES_REQUESTED** [review]
fix conflicts

File: websites/R/Roblox/dist/metadata.json [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
+    "nl": "De missie van Roblox is om de wereld samen te brengen door te spelen.",
+    "zh_CN": "Roblox的使命是通过游戏将世界融合在一起。"
+  },
+  "contributors": [
+    {
+      "name": "YJJcoolcool",
+      "id": "300929818231308288"
+    }
+  ]
 }
```
[the remark this answers is not in the document]
Comment (YJJcoolcool) [reply websites/R/Roblox/dist/metadata.json]: Sorry, I just copied and pasted the generated metadata.json from the online metadata tool. Anyway, I rearranged the placing of the description and URL to the original place before the changes and changed the version back to 2.4.0.

File: websites/R/Roblox/presence.ts (hunk 2 above)
Comment (YJJcoolcool) [reply websites/R/Roblox/presence.ts]: Yep, just checked and it's supposed to be inside updatedata event. Made the necessary changes.

**Review by Bas950 — APPROVED** [review]

**Review by StrikerFRFX — APPROVED** [review]

**Pull request merged**